### PR TITLE
Decouple `$includeThrowableDetail` from generic `debug` config

### PR DESCRIPTION
**Pull request by pascalheidmann**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

This PR allows to control the `$includeThrowableDetail` parameter to the `ProblemDetailsResponseFactoryFactory` separately from generic `debug` config (that might be reused for other things in a project). It falls back to `debug` config as it used to be and therefore should be only a minor enhancement without much to think about ;)

### CODE REVIEW

File: src/ProblemDetailsResponseFactoryFactory.php [hunk 1]
```diff
@@ -20,15 +20,22 @@ public function __invoke(ContainerInterface $container): ProblemDetailsResponseF
...
-        $includeThrowableDetail = $debug ?? ProblemDetailsResponseFactory::EXCLUDE_THROWABLE_DETAILS;
+        $debug   = isset($config['debug']) && is_bool($config['debug']) ? $config['debug'] : null;
+        $debug ??= ProblemDetailsResponseFactory::EXCLUDE_THROWABLE_DETAILS;
 
         $problemDetailsConfig = $config['problem-details'] ?? [];
         Assert::isArrayAccessible($problemDetailsConfig);
+
+        $includeThrowableDetail   = isset($problemDetailsConfig['include-throwable-details'])
+            && is_bool($problemDetailsConfig['include-throwable-details'])
+            ? $problemDetailsConfig['include-throwable-details'] : null;
```
Comment (Ocramius) [inline src/ProblemDetailsResponseFactoryFactory.php:31]: IMO OK to do this, but then let's test the additional flags being supported

**Review by Ocramius — APPROVED** [review]
LGTM. @froschdesign any opinions for docs?

File: docs/book/response.md [hunk 2]
```diff
@@ -151,6 +151,11 @@ This package also provides a factory for generating the
     value is provided as the `$defaultTypesMap` parameter; see the
     [default types documentation](default-types.md) for details on defining
     this map. (Since 1.1.0.)
+  - If the service contains a `problem-details` key with an array value
```
Comment (froschdesign) [inline docs/book/response.md:154]: The entire list without a code example is useless, but this can be changed in another pull request. But please indent the nested list with 4 spaces from "If the service contains a `debug` key...", otherwise the output is not correct.
Thanks in advance! 👍🏻

Comment (pascalheidmann) [reply docs/book/response.md:154]: Hope I got you right with that indention :)

Comment (froschdesign) [reply docs/book/response.md:154]: I will fix the Markdown linting because [MkDocs](https://www.mkdocs.org) needs some more spaces. I will check also the MkDocs configuration for this option.

- https://github.com/laminas/laminas-continuous-integration-action/blob/1.39.x/setup/markdownlint/markdownlint.json
- https://github.com/DavidAnson/markdownlint/blob/main/doc/md007.md

**Review by froschdesign — APPROVED** [review]

**Review by Ocramius — APPROVED** [review]
Thanks @pascalheidmann!
